### PR TITLE
Added compatibility with Typescript

### PR DESCRIPTION
**Pull request by jmbl1685**
Hi, @aichbauer 

I added compatibility with Typescript, the coverage and the tests are 100%.

Do the tests yourself, and notify me how it went with that, thanks for your module helped me a lot to solve something I was doing.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -17,4 +18,4 @@ export const convertArrayToCSV = (data, { header, separator } = {}) => {
   return convertArrayOfObjectsToCSV(data, thisOptions);
 };
 
-export default convertArrayToCSV;
+export { convertArrayToCSV };
```
Comment (aichbauer) [inline src/index.ts:21]: We still need the default export and the named export

File: test/fixtures/data.js [hunk 2]
```diff
@@ -89,37 +87,18 @@ const dataWithNullAndUndefined = [
   },
 ];
 
-export const dataArrayWithHeader = [
-  ...headerArray,
-  ...dataArray,
-];
+exports.dataArrayWithHeader = [...headerArray, ...dataArray];
```
Comment (aichbauer) [inline test/fixtures/data.js:90]: Why did you change the way of exporting variables?

File: test/fixtures/expected-results.js [hunk 3]
```diff
@@ -1,29 +1,67 @@
...
-export const expectedResultArrayHeaderSeparatorSemicolon = expectedResultObjectHeaderSeparatorSemicolon; // eslint-disable-line
+const expectedResultArrayHeaderSeparatorSemicolon = expectedResultObjectHeaderSeparatorSemicolon; // eslint-disable-line
 
-export const expectedResultArrayOnlySeparatorTab = expectedResultObjecOnlySeparatorTab;
+const expectedResultArrayOnlySeparatorTab = expectedResultObjecOnlySeparatorTab;
 
-export const expectedResultObjectWithDoubleQoutesInsideElement = `number,first,last,handle\n${expectedResultArrayWithDoubleQoutesInsideElement}`; // eslint-disable-line
+const expectedResultObjectWithDoubleQoutesInsideElement = `number,first,last,handle\n${expectedResultArrayWithDoubleQoutesInsideElement}`; // eslint-disable-line
+
+module.exports = {
```
Comment (aichbauer) [inline test/fixtures/expected-results.js:41]: Why changing the export again?

File: test/fixtures/options.js [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 };
 
-export const optionsHeaderDefaultSeperatorTab = {
+exports.optionsHeaderDefaultSeperatorTab = {
   header: undefined,
   separator: '\t',
 };
 
-export const optionsDefault = {
+exports.optionsDefault = {
```
Comment (aichbauer) [inline test/fixtures/options.js:21]: ?

File: test/helpers/check-if-valid.spec.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
-import { checkIfValid } from '../../src/helpers/check-if-valid';
+const { checkIfValid } = require('../../lib/helpers/check-if-valid');
```
Comment (aichbauer) [inline test/helpers/check-if-valid.spec.js:1]: Why can't we use `import` any more?

**Review by aichbauer — CHANGES_REQUESTED** [review]
We still need a way to lint our files, I think it is still possible to lint it with `eslint` right?

We need the `yarn.lock` file to be present in the repository.

File: test/fixtures/data.js (hunk 2 above)
Comment (jmbl1685) [reply test/fixtures/data.js:90]: The reason was because I removed Babel from the project. If the module passes to Typescript, I no longer need it. What you think?

File: test/fixtures/expected-results.js (hunk 3 above)
Comment (jmbl1685) [reply test/fixtures/expected-results.js:41]: Explained in the previous comment

File: src/index.ts (hunk 1 above)
Comment (jmbl1685) [reply src/index.ts:21]: Could you try the new changes? Everything worked for me and I've it working on a project.

Comment (aichbauer) [reply src/index.ts:21]: Yeah it works but we this version has a default and a named export so a major release would be required if we change that to only named. So we still need both exports.

File: test/fixtures/data.js (hunk 2 above)
Comment (aichbauer) [reply test/fixtures/data.js:90]: Yep that I could see, but once you do `export.variableName`, then `const variableName = ''` and `module.exports = { variableName };`.

Jest can run test with `import` and `export` if using babel. I would prefer it that way.

**Comment by aichbauer** [comment]
@jmbl1685 we still need a way to lint this package.
